src/invoker.ts

```diff
@@ -358,7 +360,10 @@ function makeHttpHandler(execute: HttpFunction): express.RequestHandler {
         logAndSendError(err, res);
       }
     });
-    d.run(() => {
+    d.run(async () => {
+      // Start the debugger (if not alreay started)
+      const tools = startTooling();
```

Comment:
There is overhead both in starting the tooling, and ongoing performance overhead. Is there a way for the user to opt out of this?

Reply:
For ease in turning instrumentation on and off, it would be nice to toggle this with an environment variable instead of something requiring a redeployment (e.g., adjusted container entrypoint)

Reply:
Yes. The PRD specifies the following semantics:

**Enable/disable Stackdriver APM**
As a developer, I can enable or disable individual Stackdriver APM features. The on/off state is controlled by an environment variable as follows:

Stackdriver APM feature | Environment variable
-- | --
Stackdriver Trace | GOOGLE_CLOUD_TRACE_ENABLED

If the environment variable is set to `false` or if it is not set, the feature should be disabled by default.

**By default, Stackdriver APM features are enabled on Cloud Functions**
As a developer on Cloud Functions, when I deploy my function Stackdriver Trace is enabled by default. The container into which my function is deployed will, e.g., automatically set the value of the `GOOGLE_CLOUD_TRACE_ENABLED` environment variable to `true`.